src/main/java/org/gridsuite/study/server/dto/VoltageLevelInfos.java

```diff
@@ -20,6 +21,7 @@
 @NoArgsConstructor
 @Getter
 @ToString
+@EqualsAndHashCode
```

Comment:
Alternatively, you can specify exactly which fields or methods you wish to be used by marking them with @EqualsAndHashCode.Include and using @EqualsAndHashCode(onlyExplicitlyIncluded = true). 

Then maybe only on the `'id'` attribute ?